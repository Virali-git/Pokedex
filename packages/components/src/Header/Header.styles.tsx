export const HeaderTypoStyle = {
  flexGrow: 1,
  fontWeight: "bold",
  color: "#fff",
  textShadow: "2px 2px 4px rgba(0, 0, 0, 0.3)",
  fontFamily: "Pacifico, cursive",
  fontSize: "24px",
};

export const AppBarStyle = {
  background: "linear-gradient(135deg, #ff6600 30%, #ffad33 90%)",
  boxShadow: "0px 2px 5px rgba(0, 0, 0, 0.2)",
};
