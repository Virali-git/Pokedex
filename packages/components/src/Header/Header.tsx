import React from "react";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import { styled } from "@mui/material/styles";

type PropsType = { data: string };

const StyledAppBar = styled(AppBar)(({ theme }) => ({
  background: "linear-gradient(135deg, #ff6600 30%, #ffad33 90%)", 
  boxShadow: "0px 2px 5px rgba(0, 0, 0, 0.2)", 
}));
export const Header: React.FC<PropsType> = ({ data }) => {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <StyledAppBar position="fixed">
        <Toolbar>
          <Typography
            variant="h6"
            component="div"
            sx={{
              flexGrow: 1,
              fontWeight: "bold",
              color: "#fff",
              textShadow: "2px 2px 4px rgba(0, 0, 0, 0.3)", 
              fontFamily: "Pacifico, cursive", 
              fontSize: "24px",
            }}
          >
            {data}
          </Typography>
        </Toolbar>
      </StyledAppBar>
    </Box>
  );
};
