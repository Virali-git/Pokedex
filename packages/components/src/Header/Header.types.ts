export type PropsType = { data: string };
