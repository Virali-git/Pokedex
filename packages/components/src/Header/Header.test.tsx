import React from "react";
import { render, screen } from "@testing-library/react";
import { Header } from "./Header";

describe("Header Component", () => {

  it("renders the app bar with correct props", () => {
    const  getByText  = render(<Header data="pokedex" />);

    expect(screen.getByText("pokedex")).toBeInTheDocument();
    expect(getByText).toMatchSnapshot();
  });
  it("render the correct styles required for header", () => {
    const { getByText } = render(<Header data="pokedex"/>);
    const appNameElement = getByText("pokedex");
    //testCases to check the required styles from CMS
    expect(appNameElement).toHaveStyle("font-size: 24px");
  });
});
