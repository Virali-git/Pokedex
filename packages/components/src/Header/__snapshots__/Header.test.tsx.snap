// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders the app bar with correct props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-i9gxme"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1cx9dwv-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 css-1qj1am3-MuiTypography-root"
            >
              pokedex
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-i9gxme"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1cx9dwv-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1qj1am3-MuiTypography-root"
          >
            pokedex
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
